pkg/kademlia/peer_discovery.go

```diff
@@ -117,3 +121,109 @@ func isDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+// discoveryQueue is a limited priority queue for nodes with xor distance
+type discoveryQueue struct {
+	maxLen int
+	mu     sync.Mutex
+	added  map[storj.NodeID]int
+	items  []queueItem
+}
+
+// queueItem is node with a priority
+type queueItem struct {
+	node     *pb.Node
+	priority storj.NodeID
+	// TODO: switch to using pb.NodeAddress to avoid pointer to *pb.Node
```

Comment:
We will want to store the node not the address. The node contains more information that we will need in the future than just the address.

Reply:
Discovery discovers nodes not addresses. Once we connect to a node we put the entire node in both the cache for a satellite and routing table for local nodes. We could rebuild a node but would lose any information we also have about it that passed along during discovery. i.e nodetype

Reply:
Yes, but the peer discovery wouldn't do that. Once the other node pings us back, then we add it.

Reply:
yes that is a step we take before adding them but a node that pings us back doesn't pass itself along with it.

Reply:
A ping merely verifies a node is alive at the given address and it's ID matches the certificate ID

Reply:
removed todo comment...